frontend/libraries/ballerina-core/src/forms/domains/singleton/template.tsx

```diff
@@ -34,6 +34,13 @@ export const Form = <Entity, FieldStates, Context, ForeignMutationsExpected>() =
                       modifiedByUser:true,
                       validation:Debounced.Updaters.Template.value<FormValidatorSynchronized>(Synchronized.Updaters.value(replaceWith(unit)))(_[field].validation),
                     }) }))
+                  } else {
+                    props.setState(_ => ({ ..._,
+                      modifiedByUser: true,
+                      [field]:({
+                        ..._[field],
+                        modifiedByUser:true,
+                      }) }))
```

Comment:
causes a number of renders specific to how deep the component is in the form. E.g. if its inside 3 other forms, we will cause 4 renders. This makes sense since all of those forms get marked as "modifiedByUser).